lib/internal/streams/web.js

```diff
@@ -0,0 +1,4632 @@
+    }
+    return {
+      data: {
+        readable,
+        writable,
+      },
+      deserializeInfo: 'internal/streams/whatwg:TransferedTransformStream'
+    };
+  }
+
+  [kTransferList]() {
+    return [ this[kState].readable, this[kState].writable ];
+  }
+
+  [kDeserialize]({ readable, writable }) {
+    this[kState].readable = readable;
+    this[kState].writable = writable;
+  }
+}
+
+class TransferedTransformStream extends JSTransferable {
+  constructor() {
+    super();
+    this[kType] = 'TransformStream';
+    this[kState] = {
+      readable: undefined,
+      writable: undefined,
+      backpressure: undefined,
+      backpressureChange: {
+        promise: undefined,
+        resolve: undefined,
+        reject: undefined,
+      },
+      controller: undefined,
+    };
+  }
+}
+
+class TransformStreamDefaultController {
+  [kType] = 'TransformStreamDefaultController';
+
+  constructor() {
+    throw new ERR_ILLEGAL_CONSTRUCTOR();
+  }
+
+  /**
+   * @readonly
+   * @type {number}
+   */
+  get desiredSize() {
+    if (!isTransformStreamDefaultController(this))
+      throw new ERR_INVALID_THIS('TransformStreamDefaultController');
+    const {
+      stream,
+    } = this[kState];
+    const {
+      readable,
+    } = stream[kState];
+    const {
+      controller: readableController,
+    } = readable[kState];
+    return readableStreamDefaultControllerGetDesiredSize(readableController);
+  }
+
+  /**
+   * @param {any} chunk
+   */
+  enqueue(chunk) {
+    if (!isTransformStreamDefaultController(this))
+      throw new ERR_INVALID_THIS('TransformStreamDefaultController');
+    transformStreamDefaultControllerEnqueue(this, chunk);
+  }
+
+  /**
+   * @param {any} reason
+   */
+  error(reason) {
+    if (!isTransformStreamDefaultController(this))
+      throw new ERR_INVALID_THIS('TransformStreamDefaultController');
+    transformStreamDefaultControllerError(this, reason);
+  }
+
+  terminate() {
+    if (!isTransformStreamDefaultController(this))
+      throw new ERR_INVALID_THIS('TransformStreamDefaultController');
+    transformStreamDefaultControllerTerminate(this);
+  }
+
+  [kInspect](depth, options) {
+    return customInspect(depth, options, 'TransformStreamDefaultController', {
+      stream: this[kState].stream,
+    });
+  }
+}
+
+class InternalTransformStreamDefaultController {
+  constructor() {
+    this[kType] = 'TransformStreamDefaultController';
+  }
+}
+
+class ByteLengthQueuingStrategy {
+  [kType] = 'ByteLengthQueuingStrategy';
+
+  /**
+   * @param {{
+   *   highWaterMark : number
+   * }} init
+   */
+  constructor(init) {
+    validateObject(init, 'init');
+    if (init.highWaterMark === undefined) {
+      throw new ERR_INVALID_ARG_VALUE(
+        'options.highWaterMark',
+        init.highWaterMark);
+    }
+    // The highWaterMark value is not checked until the strategy
+    // is actually used, per the spec.
+    this[kState] = {
+      highWaterMark: Number(init.highWaterMark),
```

Comment:
```suggestion
      highWaterMark: +init.highWaterMark,
```
this is identical but doesn't require calling the primordial

Reply:
I think it more readable with Number()

Reply:
I do agree, but this isn't an npm library, it's node, and syntax is better than calling a primordial. my guess is it's more performant as well.